File: src/WorkflowShell/mvn/MavenDependencyAnalysis.rb
require_relative '../models/Command'

class MavenCompile < Command
  @command_string
  @command_description
  @command_usage

  def initialize
    @command_string = 'mvndeps'
    @command_description = 'Prints a dependency report, showing used but undeclared dependencies and other useful stuff.'
    @command_usage = @command_string
  end

  def run_command(arguments)
    # Ask the Command class to parse the basic options like help and verbosity
    parser_components = parse_options(arguments, true)

    command = "mvn -U -T 3C clean dependency:analyze -Djava.awt.headless=true"
    run_shell_command(command, parser_components.basic_options.verbose)
  end
end

File: src/WorkflowShell/mvn/MavenDependencyAnalysis.ts
import Command from "../models/Command";

export default class MavenDependencyAnalysis extends Command {

  static getString(): string {
    return "mdeps";
  }

  static getDescription(): string {
    return "Prints a dependency report, showing used but undeclared dependencies and other useful stuff.";
  }

  getUsage(): string {
    return MavenDependencyAnalysis.getString();
  }

  async run(args: any) {
    const ok: boolean = await super.run(args);
    if (!ok) {
      return false;
    }

    const baseCommand: string = "mvn -U -T 3C clean dependency:analyze -Djava.awt.headless=true";
    super.execute(baseCommand);
    return true;
  }
}
